### DISCUSSION TITLE
I am unable to turn on GPU mining because it says my AMD GPU is not compatible / My AMD GPU is mining but Experimental Settings says that no GPU devices are found?

### DISCUSSION THREAD

**Discussion by SolfataraEmit** [tari-project/universe#1002] [Q&A] [answered]
I am unable to turn on GPU mining because it says my AMD GPU is not compatible / My AMD GPU is mining but Experimental Settings says that no GPU devices are found?

**Comment by stringhandler** [comment, marked as answer]
This behaviour can manifest in several ways. Your GPU power displays a zero/blank hash rate, or one that is far below what one might expect. Going into the settings, you’ll find the GPU Mining might indicate that you don’t have a compatible graphics card. Lastly, in the Experimental Settings, users might receive a message stating “No GPU devices found”

The culprit is either that your card doesn’t support OpenCL or - more likely - that the driver set you have installed doesn’t have a version of OpenCL available, or isn't properly linked.

AMD in particular has an issue in that its more recent driver install sets no longer include the OpenCL drivers and libraries. Most recent hardware does, however, support OpenCL.

## Windows

The easiest way to resolve this is to determine if your current driver set has OpenCL is to download [GPU-Z](https://www.techpowerup.com/gpuz/) and run it. In the bottom of the dialog will be a list of supported features that are currently installed. If OpenCL isn’t checked, that’s an indicator that your driver set doesn’t include the OpenCL drivers.

Our recommendation is to refer to the following repo for assistance: https://github.com/ptrumpis/OpenCL-AMD-GPU

## Mac

In the System Report, check under Software —> Frameworks to see if the OpenCL framework is installed. You can also use `brew install clinfo`, and then run it to get more detailed information.

You may need to update your MacOS, which should also install the latest drivers. If this doesn’t resolve your issue, you may need to check that the OpenCL libraries are properly linked. 

```
otool -L /System/Library/Frameworks/OpenCL.framework/OpenCL
```

This will list the libraries used by OpenCL. If you see errors or missing libraries, your macOS might have a problem with its OpenCL framework.

## Linux

You can use the tool `clinfo` to determine if your card supports OpenCL. If you have working drivers, your device should be listed when entering the command, like so:

```
Platform Name: AMD Accelerated Parallel Processing
Platform Vendor: Advanced Micro Devices, Inc.
Platform Version: OpenCL 1.2 AMD-APP (2988.4)

Device Name: AMD Radeon RX 580 Series
Device Vendor: Advanced Micro Devices, Inc.
Device Version: OpenCL 1.2
```

If your device is not listed, you are going to likely have to use the AMDGPU-PRO drivers to get the necessary support. Refer to your repos particular instructions for how to install the necessary drivers

Below is an example for Ubuntu:

```
sudo apt install opencl-amdgpu-pro-icd
```
